sample/rules.go

```diff
@@ -0,0 +1,158 @@
+package sample
+
+import (
+	"math/rand"
+	"strings"
+
+	"github.com/honeycombio/samproxy/config"
+	"github.com/honeycombio/samproxy/logger"
+	"github.com/honeycombio/samproxy/types"
+)
+
+type RulesBasedSampler struct {
+	Logger logger.Logger
+	Config *config.RulesBasedSamplerConfig
+}
+
+func (s *RulesBasedSampler) Start() error {
+	s.Logger.Debug().Logf("Starting RulesBasedSampler")
+	defer func() { s.Logger.Debug().Logf("Finished starting RulesBasedSampler") }()
+
+	return nil
+}
+
+func (s *RulesBasedSampler) GetSampleRate(trace *types.Trace) (rate uint, keep bool) {
+	for _, rule := range s.Config.Rule {
+		var matched int
+
+		// no condition signifies the default
+		if rule.Condition == nil {
+			return uint(rule.SampleRate), rand.Intn(rule.SampleRate) == 0
```

Comment:
I think `rand.Intn` will panic if n is 0, but in the test it looked like a sample rate of 0 was valid and meant that it should be dropped?

Reply:
ah good point we check for this down below but not if there is no condition. I'll add a test and fix it